### PR TITLE
Add direct display support

### PR DESCRIPTION
**Pull request by tomek-brcm**
Try vkCreateDisplayPlaneSurfaceKHR() from VK_KHR_display extension on Unix systems before falling back to the headless mode.

### CODE REVIEW

**Comment by tomek-brcm** [comment]
Rebased and updated to match the current dev branch.

File: framework/application/display_window.cpp [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+VkResult DisplayWindow::SelectDisplay(const encode::InstanceTable* table,
+                                      VkPhysicalDevice             physical_device,
+                                      VkDisplayKHR*                display) const
+{
+    assert(table);
+    uint32_t num_displays;
+    VkResult error = table->GetPhysicalDeviceDisplayPropertiesKHR(physical_device, &num_displays, nullptr);
+    if (error || !num_displays)
+        return error ? error : VK_ERROR_INITIALIZATION_FAILED;
```
Comment (patrick-lunarg) [inline framework/application/display_window.cpp:76]: Curly braces

**Review by patrick-lunarg — CHANGES_REQUESTED** [review]

File: framework/application/display_window.cpp (hunk 1 above)
Comment (tomek-brcm) [reply framework/application/display_window.cpp:76]: Hi Patrick,
Do you round every if() block?

**Comment by patrick-lunarg** [comment]
Yes

File: framework/application/display_window.cpp (hunk 1 above)
Comment (patrick-lunarg) [reply framework/application/display_window.cpp:76]: Yes

Comment (tomek-brcm) [reply framework/application/display_window.cpp:79]: done

**Review by patrick-lunarg — APPROVED** [review]

File: framework/application/display_window.cpp [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+}
+
+VkResult DisplayWindow::SelectPhysicalDevice(const encode::InstanceTable* table,
+                                             VkInstance                   instance,
+                                             VkPhysicalDevice*            physical_device) const
+{
+    assert(table);
+    uint32_t physical_device_count = 0;
+    VkResult error                 = table->EnumeratePhysicalDevices(instance, &physical_device_count, nullptr);
+    if (error || !physical_device_count)
```
Comment (bradgrantham-lunarg) [inline framework/application/display_window.cpp:44]: This test and the body are somewhat difficult for me to read.  I think they combine reporting of two conditions in the name of brevity, but I prefer it to be clearer.  Can this and other similar clauses be factored out into something like this?
```
if (error != VK_SUCCESS) {
    return error;
}
if(physical_device_count == 0)
{
    return VK_ERROR_INITIALIZATION_FAILED;
}
```

**Review by bradgrantham-lunarg — CHANGES_REQUESTED** [review]

File: framework/application/display_window.cpp (hunk 1 above)
Comment (bradgrantham-lunarg) [reply framework/application/display_window.cpp:81]: `CONTRIBUTING.md` states we try to follow the [Google Style Guide](https://google.github.io/styleguide/cppguide.html), which (unless I've misread it) prefers {} around every if and else block.

File: framework/application/display_window.cpp (hunk 2 above)
Comment (tomek-brcm) [reply framework/application/display_window.cpp]: Done.

**Comment by tomek-brcm** [comment]
@bradgrantham-lunarg The 'CI Build / Ubuntu GCC Debug' looks like a glitch in the matrix, can you re-trigger from your end?

File: framework/application/display_window.cpp (hunk 2 above)
Comment (bradgrantham-lunarg) [reply framework/application/display_window.cpp]: 👍

**Review by bradgrantham-lunarg — APPROVED** [review]

**Comment by bradgrantham-lunarg** [comment]
> @bradgrantham-lunarg The 'CI Build / Ubuntu GCC Debug' looks like a glitch in the matrix, can you re-trigger from your end?

Yes, definitely looked like a glitch.  They all pass now!

**Comment by tomek-brcm** [comment]
Is there anything else to do here?

**Comment by patrick-lunarg** [comment]
> Is there anything else to do here?

Please excuse  the delay on merging this

**Comment by tomek-brcm** [comment]
No worries, thanks for the merge!

**Pull request merged**